### PR TITLE
ROS2 Support

### PR DESCRIPTION
**Pull request by splatter96**
I updated the package with ROS2 support. For now I only updated the imu_transformer package as this is what I need at the moment and whats probably needed by most in this package.

All test cases still work as expected as the actual math is not touched only the ROS interface.

I've forgone the original approach of supporting both MagneticField and Vector3 messages for the magnetometer input as this is harder to support in ROS2 without the ShapeShifter or something similar and most IMU drivers already adopted the proposed REP 145 to use only MagneticField messages.

### CODE REVIEW

**Comment by splatter96** [comment]
@tonybaltovski  Could you take a look at this?

File: imu_transformer/package.xml [hunk 1]
```diff
@@ -14,24 +14,24 @@
   <maintainer email="pbovbel@clearpathrobotics.com">Paul Bovbel</maintainer>
   <maintainer email="tbaltovski@clearpathrobotics.com">Tony Baltovski</maintainer>
 
-  <buildtool_depend>catkin</buildtool_depend>
+  <buildtool_depend>ament_cmake</buildtool_depend>
 
   <depend>geometry_msgs</depend>
-  <depend>nodelet</depend>
+  <!--<depend>nodelet</depend>-->
```
Comment (tonybaltovski) [inline imu_transformer/package.xml]: Remove please

**Review by tonybaltovski — CHANGES_REQUESTED** [review]
LGTM with a minor nit.

**Comment by tonybaltovski** [comment]
@splatter96 looks good, just that one minor nit.

File: imu_transformer/package.xml (hunk 1 above)
Comment (splatter96) [reply imu_transformer/package.xml]: Incorporated the requested change.

**Review by tonybaltovski — APPROVED** [review]

**Pull request merged**